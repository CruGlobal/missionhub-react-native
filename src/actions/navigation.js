import { NavigationActions } from 'react-navigation';

// Performance issues: https://github.com/react-community/react-navigation/issues/608#issuecomment-328635042

export function navigatePush(screen, props = {}) {
  return (dispatch) => {
    dispatch(NavigationActions.navigate({
      routeName: screen,
      params: props,
    }));
  };
}

export function navigateBack(times, backParams) {
  return (dispatch) => {
    if (times && times > 1) {
      dispatch(NavigationActions.pop({ n: times, immediate: true }));
    } else {
      // backParams can contain { key: string, immediate: bool }
      dispatch(NavigationActions.back(backParams));
    }
  };
}

export function navigateReset(screen, props = {}) {
  return (dispatch) => {
    dispatch(NavigationActions.reset({
      index: 0,
      actions: [
        NavigationActions.navigate({ routeName: screen, params: props }),
      ],
    }));
  };
}

// The reset home and reset login are handled by the login/logout auth actions
