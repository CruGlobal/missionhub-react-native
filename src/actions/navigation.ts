/* eslint-disable @typescript-eslint/no-explicit-any, import/named */

import { AnyAction } from 'redux';
import { ThunkDispatch } from 'redux-thunk';
import {
  NavigationActions,
  StackActions,
  NavigationNavigateAction,
} from 'react-navigation';

import { FEED_ITEM_DETAIL_SCREEN } from '../containers/Communities/Community/CommunityFeedTab/FeedItemDetailScreen/FeedItemDetailScreen';
import { MAIN_TABS, PEOPLE_TAB, COMMUNITIES_TAB } from '../constants';
import { COMMUNITY_TABS } from '../containers/Communities/Community/constants';

import { loadHome } from './auth/userData';

export function navigatePush(screen: string, props = {}) {
  return (dispatch: ThunkDispatch<{}, {}, AnyAction>) => {
    dispatch(
      StackActions.push({
        routeName: screen,
        params: props,
      }),
    );
  };
}

export function navigateBack(times?: number) {
  return (dispatch: ThunkDispatch<{}, {}, AnyAction>) => {
    if (times && times > 1) {
      dispatch(StackActions.pop({ n: times, immediate: true }));
    } else {
      dispatch(NavigationActions.back());
    }
  };
}

export const navigateReset = (screen: string, props = {}) =>
  resetStack([
    NavigationActions.navigate({ routeName: screen, params: props }),
  ]);

export const navigateNestedReset = (
  screens: {
    routeName: string;
    tabName?: string;
    params?: { [key: string]: any };
  }[],
) =>
  resetStack(
    screens.map(({ routeName, params, tabName }) =>
      NavigationActions.navigate({
        routeName,
        ...(params ? { params } : {}),
        ...(tabName
          ? {
              action: NavigationActions.navigate({
                routeName: tabName,
              }),
            }
          : {}),
      }),
    ),
    screens.length - 1,
  );

const resetStack = (actions: NavigationNavigateAction[], index = 0) => (
  dispatch: ThunkDispatch<{}, {}, AnyAction>,
) =>
  dispatch(
    StackActions.reset({
      index,
      key: null, // Reset root stack navigator
      actions,
    }),
  );

// The reset home and reset login are handled by the login/logout auth actions

export function navigateReplace(screen: string, props = {}) {
  return (dispatch: ThunkDispatch<{}, {}, AnyAction>) => {
    dispatch(
      StackActions.replace({
        routeName: screen,
        params: props,
      }),
    );
  };
}

export const navigateToMainTabs = (tabName = PEOPLE_TAB) => (
  dispatch: ThunkDispatch<never, {}, AnyAction>,
) => {
  dispatch(loadHome());
  dispatch(navigateNestedReset([{ routeName: MAIN_TABS, tabName }]));
};

export const navigateToFeedItemComments = (
  feedItemId: string,
  communityId: string,
) => (dispatch: ThunkDispatch<{}, null, AnyAction>) => {
  dispatch(
    navigateNestedReset([
      {
        routeName: MAIN_TABS,
        tabName: COMMUNITIES_TAB,
      },
      {
        routeName: COMMUNITY_TABS,
        params: { communityId },
      },
      {
        routeName: FEED_ITEM_DETAIL_SCREEN,
        params: { feedItemId },
      },
    ]),
  );
};
