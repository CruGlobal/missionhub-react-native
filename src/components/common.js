// Common components that we can import with simple destructuring syntax
// So when one of these components are loaded in, all files will be called also
import Button from './Button';
import DateComponent from './DateComponent';
import Flex from './Flex';
import Icon from './Icon';
import Input from './Input';
import Loading from './Loading';
import Text from './Text';
import Touchable from './Touchable';
import Separator from './Separator';
import RefreshControl from './RefreshControl';

export {
  Button,
  DateComponent,
  Flex,
  Icon,
  Input,
  Loading,
  Text,
  Touchable,
  Separator,
  RefreshControl,
};
