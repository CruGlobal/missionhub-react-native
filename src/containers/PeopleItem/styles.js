import { StyleSheet } from 'react-native';

import theme from '../../theme';

export default StyleSheet.create({
  row: {
    paddingVertical: 15,
    paddingHorizontal: 24,
    backgroundColor: theme.white,
    width: theme.fullWidth,
    borderBottomWidth: theme.separatorHeight,
    borderBottomColor: theme.separatorColor,
  },
  name: {
    fontSize: 16,
    fontWeight: 'bold',
    color: theme.primaryColor,
  },
  stage: {
    fontSize: 14,
    color: theme.inactiveColor,
  },
  uncontacted: {
    color: theme.red,
  },
  uncontactedIcon: {
    fontSize: 24,
    color: theme.red,
  },
});
