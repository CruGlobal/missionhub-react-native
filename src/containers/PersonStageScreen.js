import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import PathwayStageScreen from './PathwayStageScreen';
import { selectPersonStage, updateUserStage } from '../actions/selectStage';
import { navigatePush, navigateBack } from '../actions/navigation';

class PersonStageScreen extends Component {
  constructor(props) {
    super(props);

    this.handleSelectStage = this.handleSelectStage.bind(this);
  }

  handleSelectStage(stage) {
    if (this.props.onComplete) {
      this.props.dispatch(updateUserStage(this.props.contactAssignmentId, stage.id)).then(()=>{
        this.props.onComplete(stage);
        this.props.dispatch(navigateBack());
      });
    } else {
      this.props.dispatch(selectPersonStage(this.props.contactId || this.props.personId, this.props.myId, stage.id)).then(() => {
        this.props.dispatch(navigatePush('PersonStep'));
      });
    }
  }

  render() {
    const questionText = `Which stage best describes where ${this.props.name || this.props.personFirstName} is on their journey?`;

    return (
      <PathwayStageScreen
        buttonText="HERE"
        questionText={questionText}
        onSelect={this.handleSelectStage} />
    );
  }

}


PersonStageScreen.propTypes = {
  onComplete: PropTypes.func,
  name: PropTypes.string,
  contactId: PropTypes.string,
  currentStage: PropTypes.string,
  contactAssignmentId: PropTypes.string,
};

const mapStateToProps = ({ personProfile, auth }, { navigation }) => ({
  ...(navigation.state.params || {}),
  personFirstName: personProfile.personFirstName,
  personId: personProfile.id,
  myId: auth.personId,
});

export default connect(mapStateToProps)(PersonStageScreen);
