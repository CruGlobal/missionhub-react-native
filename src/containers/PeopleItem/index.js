import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { translate } from 'react-i18next';

import { Flex, Text, Touchable, Icon } from '../../components/common';
import { navigatePush } from '../../actions/navigation';
import { getMyPeople } from '../../actions/people';
import { PERSON_STAGE_SCREEN } from '../PersonStageScreen';
import { isMissionhubUser, communityIsCru } from '../../utils/common';

import styles from './styles';

@translate()
export class PeopleItem extends Component {
  constructor(props) {
    super(props);

    this.state = {
      isMe: props.person.id === props.me.id,
      isPersonal: props.organization && props.organization.id === 'personal',
    };
  }

  handleSelect = () =>
    this.props.onSelect(this.props.person, this.props.organization);

  handleChangeStage = () => {
    const { me, dispatch, person, organization } = this.props;

    const contactAssignment = (person.reverse_contact_assignments || []).find(
      a => a.assigned_to.id === me.id,
    );
    const contactAssignmentId = contactAssignment && contactAssignment.id;

    dispatch(
      navigatePush(PERSON_STAGE_SCREEN, {
        onComplete: () => dispatch(getMyPeople()),
        currentStage: null,
        name: person.first_name,
        contactId: person.id,
        contactAssignmentId: contactAssignmentId,
        section: 'people',
        subsection: 'person',
        orgId: organization.id,
      }),
    );
  };

  render() {
    const { person, me, t, stagesObj, organization } = this.props;
    const { isPersonal, isMe } = this.state;
    const newPerson = isMe ? me : person;
    let personName = isMe ? t('me') : newPerson.full_name || '';
    personName = personName.toUpperCase();

    let stage = null;

    const contactAssignments = person.reverse_contact_assignments || [];
    if (isMe) {
      stage = me.stage;
    } else if (stagesObj) {
      const contactAssignment = contactAssignments.find(
        a => a.assigned_to.id === me.id,
      );
      if (
        contactAssignment &&
        contactAssignment.pathway_stage_id &&
        stagesObj[`${contactAssignment.pathway_stage_id}`]
      ) {
        stage = stagesObj[`${contactAssignment.pathway_stage_id}`];
      }
    }

    const isCruOrg = communityIsCru(organization);

    let status = 'uncontacted';

    const orgPermissions = person.organizational_permissions || [];
    if (isMe || !isCruOrg) {
      status = '';
    } else if (isCruOrg) {
      const personOrgPermissions = orgPermissions.find(
        o => o.organization_id === organization.id,
      );
      if (personOrgPermissions) {
        if (isMissionhubUser(personOrgPermissions)) {
          status = '';
        } else {
          status = personOrgPermissions.followup_status || '';
        }
      }
    }

    const isUncontacted = status === 'uncontacted';

    return (
      <Touchable highlight={true} onPress={this.handleSelect}>
        <Flex direction="row" align="center" style={styles.row}>
          <Flex justify="center" value={1}>
            <Text style={styles.name}>{personName}</Text>
            <Flex direction="row" align="center">
              <Text style={styles.stage}>{stage ? stage.name : ''}</Text>
              <Text style={styles.stage}>
                {stage && status ? '  >  ' : null}
              </Text>
              <Text
                style={[
                  styles.stage,
                  isUncontacted ? styles.uncontacted : null,
                ]}
              >
                {t(status ? `followupStatus.${status.toLowerCase()}` : null)}
              </Text>
            </Flex>
          </Flex>
          {!isPersonal && !stage && !isMe ? (
            <Touchable isAndroidOpacity={true} onPress={this.handleChangeStage}>
              <Icon
                name="journeyIcon"
                type="MissionHub"
                style={styles.uncontactedIcon}
              />
            </Touchable>
          ) : null}
        </Flex>
      </Touchable>
    );
  }
}

PeopleItem.propTypes = {
  person: PropTypes.shape({
    id: PropTypes.string.isRequired,
    full_name: PropTypes.string.isRequired,
    first_name: PropTypes.string,
    last_name: PropTypes.string,
    gender: PropTypes.string,
    student_status: PropTypes.string,
    campus: PropTypes.string,
    year_in_school: PropTypes.string,
    major: PropTypes.string,
    minor: PropTypes.string,
    birth_date: PropTypes.string,
    date_became_christian: PropTypes.string,
    graduation_date: PropTypes.string,
    picture: PropTypes.string,
    created_at: PropTypes.string,
    updated_at: PropTypes.string,
  }).isRequired,
  onSelect: PropTypes.func.isRequired,
  organization: PropTypes.object,
};

const mapStateToProps = ({ auth, stages }) => ({
  me: auth.person,
  stagesObj: stages.stagesObj,
});

export default connect(mapStateToProps)(PeopleItem);
