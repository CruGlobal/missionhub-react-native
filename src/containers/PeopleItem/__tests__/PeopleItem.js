import 'react-native';
import React from 'react';
import { Provider } from 'react-redux';

import {
  createMockStore,
  renderShallow,
  testSnapshot,
} from '../../../../testUtils';

import PeopleItem from '..';

import { navigatePush } from '../../../actions/navigation';
import { PERSON_STAGE_SCREEN } from '../../PersonStageScreen';
import { orgIsCru, isMissionhubUser } from '../../../utils/common';

const myId = '1';
const stageId = '1';

const mockStages = {
  [stageId]: {
    id: stageId,
    name: 'Stage 1',
  },
};

const mockState = {
  auth: {
    person: {
      id: myId,
      stage: mockStages[stageId],
    },
  },
  stages: {
    stagesObj: mockStages,
  },
};

const store = createMockStore(mockState);

const onSelect = jest.fn();

const mockPersonalMinistry = { id: 'personal' };
const mockOrganization = { id: '1', user_created: false };

const mockOrgPermission = {
  organization_id: mockOrganization.id,
  followup_status: 'contacted',
};
const mockOrgPermissionUncontacted = {
  organization_id: mockOrganization.id,
  followup_status: 'uncontacted',
};
const mockContactAssignment = {
  id: '90',
  assigned_to: { id: myId },
  pathway_stage_id: '1',
};
const mockContactAssignmentNoStage = {
  id: '90',
  assigned_to: { id: myId },
};
const mockPerson = {
  id: '123',
  first_name: 'John',
  last_name: 'Doe',
  full_name: 'John Doe',
  organizational_permissions: [mockOrgPermission],
  reverse_contact_assignments: [mockContactAssignment],
};

const mockNavigatePushResult = { type: 'navigated' };
const mockGetPeopleResult = { type: 'got people' };

let shallowScreen;

jest.mock('../../../actions/people', () => ({
  getMyPeople: () => mockGetPeopleResult,
}));
jest.mock('../../../actions/navigation', () => ({
  navigatePush: jest.fn((_, params) => {
    params.onComplete();
    return mockNavigatePushResult;
  }),
}));
jest.mock('../../../utils/common');

beforeEach(() => {
  jest.clearAllMocks();
});

it('renders me correctly', () => {
  const mePerson = { ...mockPerson, id: myId };

  orgIsCru.mockReturnValue(false);
  isMissionhubUser.mockReturnValue(false);

  testSnapshot(
    <Provider store={store}>
      <PeopleItem
        onSelect={onSelect}
        person={mePerson}
        organization={mockPersonalMinistry}
      />
    </Provider>,
  );
  expect(orgIsCru).toHaveBeenCalledWith(mockPersonalMinistry);
  expect(isMissionhubUser).not.toHaveBeenCalled();
});

it('renders personal ministry contact correctly', () => {
  orgIsCru.mockReturnValue(false);
  isMissionhubUser.mockReturnValue(false);

  testSnapshot(
    <Provider store={store}>
      <PeopleItem
        onSelect={onSelect}
        person={mockPerson}
        organization={mockPersonalMinistry}
      />
    </Provider>,
  );
  expect(orgIsCru).toHaveBeenCalledWith(mockPersonalMinistry);
  expect(isMissionhubUser).not.toHaveBeenCalled();
});

it('renders cru org contact correctly', () => {
  orgIsCru.mockReturnValue(true);
  isMissionhubUser.mockReturnValue(false);

  testSnapshot(
    <Provider store={store}>
      <PeopleItem
        onSelect={onSelect}
        person={mockPerson}
        organization={mockOrganization}
      />
    </Provider>,
  );
  expect(orgIsCru).toHaveBeenCalledWith(mockOrganization);
  expect(isMissionhubUser).toHaveBeenCalledWith(mockOrgPermission);
});

it('renders cru org contact without stage correctly', () => {
  orgIsCru.mockReturnValue(true);
  isMissionhubUser.mockReturnValue(false);

  testSnapshot(
    <Provider store={store}>
      <PeopleItem
        onSelect={onSelect}
        person={{
          ...mockPerson,
          reverse_contact_assignments: [mockContactAssignmentNoStage],
        }}
        organization={mockOrganization}
      />
    </Provider>,
  );
  expect(orgIsCru).toHaveBeenCalledWith(mockOrganization);
  expect(isMissionhubUser).toHaveBeenCalledWith(mockOrgPermission);
});

it('renders uncontacted cru org contact correctly', () => {
  orgIsCru.mockReturnValue(true);
  isMissionhubUser.mockReturnValue(false);

  testSnapshot(
    <Provider store={store}>
      <PeopleItem
        onSelect={onSelect}
        person={{
          ...mockPerson,
          organizational_permissions: [mockOrgPermissionUncontacted],
        }}
        organization={mockOrganization}
      />
    </Provider>,
  );
  expect(orgIsCru).toHaveBeenCalledWith(mockOrganization);
  expect(isMissionhubUser).toHaveBeenCalledWith(mockOrgPermissionUncontacted);
});

it('renders cru org member correctly', () => {
  orgIsCru.mockReturnValue(true);
  isMissionhubUser.mockReturnValue(true);

  testSnapshot(
    <Provider store={store}>
      <PeopleItem
        onSelect={jest.fn()}
        person={mockPerson}
        organization={mockOrganization}
      />
    </Provider>,
  );
  expect(orgIsCru).toHaveBeenCalledWith(mockOrganization);
  expect(isMissionhubUser).toHaveBeenCalledWith(mockOrgPermission);
});

describe('handleChangeStage', () => {
  it('navigates to person stage screen', () => {
    shallowScreen = renderShallow(
      <PeopleItem
        onSelect={onSelect}
        person={{
          ...mockPerson,
          reverse_contact_assignments: [mockContactAssignmentNoStage],
        }}
        organization={mockOrganization}
      />,
      store,
    );

    shallowScreen
      .childAt(0)
      .childAt(1)
      .props()
      .onPress();

    expect(store.dispatch).toHaveBeenCalledWith(mockGetPeopleResult);
    expect(navigatePush).toHaveBeenCalledWith(PERSON_STAGE_SCREEN, {
      onComplete: expect.anything(),
      currentStage: null,
      name: mockPerson.first_name,
      contactId: mockPerson.id,
      contactAssignmentId: mockContactAssignment.id,
      section: 'people',
      subsection: 'person',
      orgId: mockOrganization.id,
    });
    expect(store.dispatch).toHaveBeenCalledWith(mockNavigatePushResult);
  });
});

describe('item selected', () => {
  it('calls onSelect', () => {
    shallowScreen = renderShallow(
      <PeopleItem
        onSelect={onSelect}
        person={mockPerson}
        organization={mockOrganization}
      />,
      store,
    );

    shallowScreen.props().onPress();

    expect(onSelect).toHaveBeenCalledWith(mockPerson, mockOrganization);
  });
});
