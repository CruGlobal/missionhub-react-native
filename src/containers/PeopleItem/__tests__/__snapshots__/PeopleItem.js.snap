// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders cru org contact correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "borderBottomColor": "#ECEEF2",
          "borderBottomWidth": 1,
          "paddingHorizontal": 24,
          "paddingVertical": 15,
          "width": 750,
        },
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {},
          Object {
            "flex": 1,
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#505256",
            },
            Object {
              "fontFamily": "SourceSansPro-Regular",
            },
            Array [
              Object {
                "color": "#007398",
                "fontSize": 16,
                "fontWeight": "bold",
              },
              null,
            ],
          ]
        }
      >
        JOHN DOE
      </Text>
      <View
        style={
          Array [
            Object {},
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#505256",
              },
              Object {
                "fontFamily": "SourceSansPro-Regular",
              },
              Object {
                "color": "#A0A2A6",
                "fontSize": 14,
              },
            ]
          }
        >
          Stage 1
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#505256",
              },
              Object {
                "fontFamily": "SourceSansPro-Regular",
              },
              Object {
                "color": "#A0A2A6",
                "fontSize": 14,
              },
            ]
          }
        >
            &gt;  
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#505256",
              },
              Object {
                "fontFamily": "SourceSansPro-Regular",
              },
              Array [
                Object {
                  "color": "#A0A2A6",
                  "fontSize": 14,
                },
                null,
              ],
            ]
          }
        >
          Contacted
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders cru org contact without stage correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "borderBottomColor": "#ECEEF2",
          "borderBottomWidth": 1,
          "paddingHorizontal": 24,
          "paddingVertical": 15,
          "width": 750,
        },
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {},
          Object {
            "flex": 1,
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#505256",
            },
            Object {
              "fontFamily": "SourceSansPro-Regular",
            },
            Array [
              Object {
                "color": "#007398",
                "fontSize": 16,
                "fontWeight": "bold",
              },
              null,
            ],
          ]
        }
      >
        JOHN DOE
      </Text>
      <View
        style={
          Array [
            Object {},
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#505256",
              },
              Object {
                "fontFamily": "SourceSansPro-Regular",
              },
              Object {
                "color": "#A0A2A6",
                "fontSize": 14,
              },
            ]
          }
        >
          
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#505256",
              },
              Object {
                "fontFamily": "SourceSansPro-Regular",
              },
              Object {
                "color": "#A0A2A6",
                "fontSize": 14,
              },
            ]
          }
        />
        <Text
          style={
            Array [
              Object {
                "color": "#505256",
              },
              Object {
                "fontFamily": "SourceSansPro-Regular",
              },
              Array [
                Object {
                  "color": "#A0A2A6",
                  "fontSize": 14,
                },
                null,
              ],
            ]
          }
        >
          Contacted
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Array [
              Object {
                "color": "#ffffff",
              },
              Object {
                "fontSize": 18,
              },
              Object {
                "color": "#FF5532",
                "fontSize": 24,
              },
            ],
            Object {
              "fontFamily": "icomoon",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        î¤–
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders cru org member correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "borderBottomColor": "#ECEEF2",
          "borderBottomWidth": 1,
          "paddingHorizontal": 24,
          "paddingVertical": 15,
          "width": 750,
        },
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {},
          Object {
            "flex": 1,
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#505256",
            },
            Object {
              "fontFamily": "SourceSansPro-Regular",
            },
            Array [
              Object {
                "color": "#007398",
                "fontSize": 16,
                "fontWeight": "bold",
              },
              null,
            ],
          ]
        }
      >
        JOHN DOE
      </Text>
      <View
        style={
          Array [
            Object {},
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#505256",
              },
              Object {
                "fontFamily": "SourceSansPro-Regular",
              },
              Object {
                "color": "#A0A2A6",
                "fontSize": 14,
              },
            ]
          }
        >
          Stage 1
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#505256",
              },
              Object {
                "fontFamily": "SourceSansPro-Regular",
              },
              Object {
                "color": "#A0A2A6",
                "fontSize": 14,
              },
            ]
          }
        />
        <Text
          style={
            Array [
              Object {
                "color": "#505256",
              },
              Object {
                "fontFamily": "SourceSansPro-Regular",
              },
              Array [
                Object {
                  "color": "#A0A2A6",
                  "fontSize": 14,
                },
                null,
              ],
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders me correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "borderBottomColor": "#ECEEF2",
          "borderBottomWidth": 1,
          "paddingHorizontal": 24,
          "paddingVertical": 15,
          "width": 750,
        },
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {},
          Object {
            "flex": 1,
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#505256",
            },
            Object {
              "fontFamily": "SourceSansPro-Regular",
            },
            Array [
              Object {
                "color": "#007398",
                "fontSize": 16,
                "fontWeight": "bold",
              },
              null,
            ],
          ]
        }
      >
        ME
      </Text>
      <View
        style={
          Array [
            Object {},
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#505256",
              },
              Object {
                "fontFamily": "SourceSansPro-Regular",
              },
              Object {
                "color": "#A0A2A6",
                "fontSize": 14,
              },
            ]
          }
        >
          Stage 1
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#505256",
              },
              Object {
                "fontFamily": "SourceSansPro-Regular",
              },
              Object {
                "color": "#A0A2A6",
                "fontSize": 14,
              },
            ]
          }
        />
        <Text
          style={
            Array [
              Object {
                "color": "#505256",
              },
              Object {
                "fontFamily": "SourceSansPro-Regular",
              },
              Array [
                Object {
                  "color": "#A0A2A6",
                  "fontSize": 14,
                },
                null,
              ],
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders personal ministry contact correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "borderBottomColor": "#ECEEF2",
          "borderBottomWidth": 1,
          "paddingHorizontal": 24,
          "paddingVertical": 15,
          "width": 750,
        },
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {},
          Object {
            "flex": 1,
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#505256",
            },
            Object {
              "fontFamily": "SourceSansPro-Regular",
            },
            Array [
              Object {
                "color": "#007398",
                "fontSize": 16,
                "fontWeight": "bold",
              },
              null,
            ],
          ]
        }
      >
        JOHN DOE
      </Text>
      <View
        style={
          Array [
            Object {},
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#505256",
              },
              Object {
                "fontFamily": "SourceSansPro-Regular",
              },
              Object {
                "color": "#A0A2A6",
                "fontSize": 14,
              },
            ]
          }
        >
          Stage 1
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#505256",
              },
              Object {
                "fontFamily": "SourceSansPro-Regular",
              },
              Object {
                "color": "#A0A2A6",
                "fontSize": 14,
              },
            ]
          }
        />
        <Text
          style={
            Array [
              Object {
                "color": "#505256",
              },
              Object {
                "fontFamily": "SourceSansPro-Regular",
              },
              Array [
                Object {
                  "color": "#A0A2A6",
                  "fontSize": 14,
                },
                null,
              ],
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders uncontacted cru org contact correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "borderBottomColor": "#ECEEF2",
          "borderBottomWidth": 1,
          "paddingHorizontal": 24,
          "paddingVertical": 15,
          "width": 750,
        },
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {},
          Object {
            "flex": 1,
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#505256",
            },
            Object {
              "fontFamily": "SourceSansPro-Regular",
            },
            Array [
              Object {
                "color": "#007398",
                "fontSize": 16,
                "fontWeight": "bold",
              },
              null,
            ],
          ]
        }
      >
        JOHN DOE
      </Text>
      <View
        style={
          Array [
            Object {},
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#505256",
              },
              Object {
                "fontFamily": "SourceSansPro-Regular",
              },
              Object {
                "color": "#A0A2A6",
                "fontSize": 14,
              },
            ]
          }
        >
          Stage 1
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#505256",
              },
              Object {
                "fontFamily": "SourceSansPro-Regular",
              },
              Object {
                "color": "#A0A2A6",
                "fontSize": 14,
              },
            ]
          }
        >
            &gt;  
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#505256",
              },
              Object {
                "fontFamily": "SourceSansPro-Regular",
              },
              Array [
                Object {
                  "color": "#A0A2A6",
                  "fontSize": 14,
                },
                Object {
                  "color": "#FF5532",
                },
              ],
            ]
          }
        >
          Uncontacted
        </Text>
      </View>
    </View>
  </View>
</View>
`;
